docs/polkadot/implementors-guide/guide.md

```diff
@@ -989,6 +989,67 @@ Dispatch a `PovFetchSubsystemMessage(relay_parent, candidate_hash, sender)` and
 
 (TODO: send statements to Statement Distribution subsystem, handle shutdown signal from candidate backing subsystem)
 
+### Statement Gossip Subsystem
+
+The statement gossip subsystem is responsible for gossiping out and receiving [Signed Statements](#Signed-statement-type) from validators.
+
+This subsystem communicates with the overseer via [two message types](#Candidate-Backing-Subsystem-Messages), one for overseer -> subsystem and another for subsystem -> overseer. Currently this just differentiates between statements that the subsystem has been instructed to gossip out, and statements that the subsystem has received.
+
+Here is one design for how it might look in code:
+
+```rust
+struct StatementGossipSubsystem<GS: GossipService, SP: Spawn> {
+    /// This trait comes from the legacy networking code, so it is likely to be changed at some point.
+    gossip_service: GS,
+    /// A map of relay parent hashes to the associated stop signal, and a mono-directional channel for communication.
+    jobs: HashMap<Hash, (exit_future::Signal, Sender<SignedStatement>)>,
+    /// A sender of messages to the overseer. This is not used directly, but passed to and used inside of jobs.
+    overseer_sender: Sender<StatementGossipSubsystemOut>,
+    /// A `futures` Spawner such as `sc_service::SpawnTaskHandle`.
+    spawner: SP
+}
+```
+
+On `OverseerSignal::StartWork`:
+1) `Clone` the `gossip_service` and `overseer_sender`.
+2) Create an associated `exit_future::Signal` and `exit_future::Exit`.
+3) Create a channel for subsystem -> job communication.
+4) Call `statement_gossip_job` and spawn the future using the `spawner`.
+5) Insert the exit signal and subsystem sender into `jobs`.
+
+On `OverseerSignal::StopWork`:
+1) Remove the signal and sender from `jobs`.
+2) Call `signal.fire()`, causing the future to complete.
+
+On `StatementGossipSubsystemMessageIn::StatementToGossip`:
+1) Get the associated sender for the `relay_parent` hash.
+2) Send the signed statement on it.
+
+```rust
+async fn statement_gossip_job<GS: GossipService>(
+    gossip_service: GS,
+    // The relay parent that this job is running for.
+    relay_parent: Hash,
```

Comment:
I'd expect:  A backing attestation would've relay_parent matching the candidate receipt I guess.  An availability attestation would've relay patent when the parablock gets declared backed, so well after the relay_parent of the candidate receipt.  An approval/validity attestation would've relay patent when the parablock gets declared available, even further after the relay_parent of the candidate receipt.

Reply:
Yes on the backing statements, although this module is not responsible at the moment for availability or secondary check gossip in any way. Availability bitfields could reasonably be merged with this module, but secondary check results probably will use a separate mechanism as they are likely to just be transactions circulated by the transaction queue gossip.

Reply:
I see, so those happen elsewhere probably